### PR TITLE
[DOCS] Added deploy mode info

### PR DESCRIPTION
**Pull request by hanna-liashchuk**
### _Why are the changes needed?_

image

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_
No

### CODE REVIEW

File: docs/deployment/engine_on_kubernetes.md [hunk 1]
```diff
@@ -36,6 +36,19 @@ Spark on Kubernetes config master by using a special format.
 
 You can use cmd `kubectl cluster-info` to get api-server host and port.
 
+### Deploy Mode
+
+Kyuubi supports cluster and client deploy modes.
+
+To use client deploy mode specify:
+* spark.driver.host (headless service hostname of the pod you want to use as a driver)
+* spark.kubernetes.driver.pod.name (the name of the pod)
```
Comment (pan3793) [inline docs/deployment/engine_on_kubernetes.md:45]: Emm, why should we set it in client mode?

Comment (hanna-liashchuk) [reply docs/deployment/engine_on_kubernetes.md:45]: no reason, just listing all the options here

Comment (hanna-liashchuk) [reply docs/deployment/engine_on_kubernetes.md:45]: I can add a line that cluster mode is recommended, if there is no particular reason to use client
wdyt?

Comment (pan3793) [reply docs/deployment/engine_on_kubernetes.md:45]: IMO, the client mode is out-of-box. We just need to mention that it's Kyuubi also supports cluster mode and it's always recommended to use cluster mode on K8s, the minimum required configurations are ...

Comment (hanna-liashchuk) [reply docs/deployment/engine_on_kubernetes.md]: updated

File: docs/deployment/engine_on_kubernetes.md [hunk 2]
```diff
@@ -36,6 +36,18 @@ Spark on Kubernetes config master by using a special format.
 
 You can use cmd `kubectl cluster-info` to get api-server host and port.
 
+### Deploy Mode
+
+One of the main advantages of the Kyuubi server compared to other interactive Spark clients is that it supports cluster deploy mode.
+ It is highly recommended to run Spark in k8s in cluster mode. 
+ 
+ The minimum required configurations are:
```
Comment (pan3793) [inline docs/deployment/engine_on_kubernetes.md:44]:
```suggestion
It is highly recommended to run Spark in k8s in cluster mode. 
 
The minimum required configurations are:
```

**Review by pan3793 — APPROVED** [review]

**Review by cxzl25 — COMMENTED** [review]
You can run `mvn spotless:apply` to fix the CI DOC style. @hanna-liashchuk

**Review by cxzl25 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master/1.8

**Pull request closed without merging**